app/components/Views/Settings/NetworksSettings/NetworkSettings/CustomNetworkView/CustomNetwork.test.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { backgroundState } from '../../../../../../util/test/initial-root-state';
+import renderWithProvider from '../../../../../../util/test/renderWithProvider';
+import CustomNetwork from './CustomNetwork';
+import { CustomNetworkProps } from './CustomNetwork.types';
+import { PopularList } from '../../../../../../util/networks/customNetworks';
+
+const getMockState = () => ({ engine: { backgroundState } });
+
+jest.mock('@react-navigation/native', () => {
+  const actualNav = jest.requireActual('@react-navigation/native');
+  return {
+    ...actualNav,
+    useNavigation: () => ({
+      navigate: jest.fn(),
+    }),
+  };
+});
+
+describe('CustomNetwork component', () => {
+  const getMockCustomNetworkProps = () => {
+    const mockCloseNetworkModal = jest.fn();
+    const mockShowNetworkModal = jest.fn();
+    const mockCustomNetworkProps: CustomNetworkProps = {
+      showPopularNetworkModal: false,
+      isNetworkModalVisible: false,
+      closeNetworkModal: mockCloseNetworkModal,
+      selectedNetwork: {
+        chainId: '0x1',
+        nickname: 'Ethereum Mainnet',
+        rpcUrl: 'https://mainnet.infura.io/v3/your-api-key',
+        ticker: 'ETH',
+        rpcPrefs: {
+          blockExplorerUrl: 'https://etherscan.io',
+        },
+      },
+      showNetworkModal: mockShowNetworkModal,
+      shouldNetworkSwitchPopToWallet: false,
+    };
+    return mockCustomNetworkProps;
+  };
+
+  it('filters out CAIP-2 networks', () => {
+    const props = getMockCustomNetworkProps();
+    const mockState = getMockState();
+    const { getByText, queryByText } = renderWithProvider(
+      <CustomNetwork {...props} showAddedNetworks />,
```

Comment:
S: have showAddedNetworks defined in mock with other props, for readability.
**Suggestion part 2/2**
```suggestion
      <CustomNetwork {...props} />,
```

Reply:
Using a different approach in the [above comment](https://github.com/MetaMask/metamask-mobile/pull/15283/files/8f9dadbcab7a4fe316c6c03e68babbb3eac314ec#r2088595338).

Using `<CustomNetwork {...props} />` by adding the `showAddedNetworks` to default props makes the test less readable and more confusing (it is hard to see why these set of props are important).

Instead we will specify the props used to trigger the test.
```ts
const props = getMockCustomNetworkProps({ showAddedNetworks: true })
const {} = render(<CustomNetwork {...props} />)
```

Reply:
Hmm I'll also update the test name, to indicate that showAddedNetworks are important to show the added networks in the component we are testing.

Reply:
Fixed in https://github.com/MetaMask/metamask-mobile/pull/15283/commits/3a84fb35a75f6d60978e5fd1651ddfdae77a2898